packages/expo-image/src/ExpoImage.tsx

```diff
@@ -1,12 +1,18 @@
 import React from 'react';
-import { Image, requireNativeComponent } from 'react-native';
+import { Image, requireNativeComponent, StyleSheet } from 'react-native';
 
 import { ImageProps } from './Image';
 
 const NativeExpoImage = requireNativeComponent('ExpoImage');
 
-export default function ExpoImage({ source, ...props }: ImageProps) {
+export default function ExpoImage({ source, style, ...props }: ImageProps) {
   const resolvedSource = Image.resolveAssetSource(source ?? {});
+  let resolvedStyle = style;
 
-  return <NativeExpoImage {...props} source={resolvedSource} />;
+  if (!Array.isArray(resolvedSource)) {
+    const { width, height } = resolvedSource;
+    resolvedStyle = StyleSheet.flatten([{ width, height }, style]);
```

Comment:
This code was taken primarily from the RN source:
https://github.com/facebook/react-native/blob/master/Libraries/Image/Image.ios.js#L103

I believe that it is used for Yoga layout purposes only and it only applies to `require(..)` assets. These assets contain the width/height and form the default when laying out the image component.

Reply:
Would there be any difference if we didn't provide the `width` and `height`? I'm afraid polyfilling this makes a difference in how different images types are handled and rendered which is something I think we want to avoid. 🤔

Reply:
I will take a closer look! 👓

Reply:
Okay. So it turns out my suspicion was right here. The `{width, height}` styles are passed down to the Yoga layout system so it can use them as the `intrinsic size` of the image. This means that when you omit any size styles, then the View will use the size of the image itself. 

I've takes a look at doing this at the native level, and it can be done. However, because of the way RN implements this, it has to schedule a block on the UI-manager (which may cause it to wait), and it always causes the layout to be invalidated, causing unnecessary layout passes.
Doing this in JS is really the best performing way as the size information is passed to Yoga directly. I've added comments to the code elaborating on this.

Reply:
I see… my personal opinion on that would be to treat all images alike, so either no user-set size = no image (intrinsic size `(0, 0)`) or to always use image's size as intrinsic size; I think this would be an improvement over current RN's API.

Let's discuss it (its feasibility) in the API proposal (soon to come)! 😃 

Until we decide otherwise I agree with you on the `{width, height}` passed to Yoga. 👍

Reply:
Let's do that 👍